src/components/Table/GridTable.tsx

```diff
@@ -944,40 +888,56 @@ function GridRow<R extends Kinded, S>(props: GridRowProps<R, S>): ReactElement {
   // We treat the "header" kind as special for "good defaults" styling
   const isHeader = row.kind === "header";
   const rowStyle = rowStyles?.[row.kind];
+  const Row = as === "table" ? "tr" : "div";
+
+  const openCardStyles =
+    typeof openCards === "string"
+      ? openCards
+          .split(":")
+          .map((openCardKind) => style.nestedCards!.kinds[openCardKind])
+          .filter((style) => style)
+      : undefined;
+
+  const leafCardStyles = isLeafRow(row) ? style.nestedCards?.kinds[row.kind] : undefined;
+  // Calculate the horizontal space already allocated by the open cards (paddings and borders)
+  const openCardWidth = openCardStyles ? openCardStyles.reduce((acc, o) => acc + o.pxPx + (o.bColor ? 1 : 0), 0) : 0;
+  // Subtract the openCardWidth from the `firstLastColumnWidth` to determine the amount of padding to add to this card.
+  // Also if it is a leaf card, then we need to additionally subtract out the border width to have it properly line up with the chrome rows
+  const currentCardPaddingWidth =
+    (style.nestedCards?.firstLastColumnWidth ?? 0) - openCardWidth - (leafCardStyles?.bColor ? 1 : 0);
```

Comment:
Do you think we could extract this to a `getNestedCardStyles(openCardStyles, row, style)` that lived in `nestedCards.tsx` and we could potentially even unit test?

And then lines 921 to 934 would live in this method, and line 921 would turn into `...getNestedCardStyles(openCardStyles, row, style)` or something like that. Maybe do that conditionally, but then ideally it's one/maybe two conditionals here in `GridRow`, and most of the card/leaf-specific local variables and other conditionals are hidden instead of `getNestedCardStyles`...